### PR TITLE
Triangle Facts jnyman

### PR DESCRIPTION
**Pull request by jnyman**
This is my refactor attempt at the Triangle Facts code example.

My focus was on trying to practice the principles of DRY and SRP.

### CODE REVIEW

File: triangle_facts/triangle_facts.rb [hunk 1]
```diff
@@ -1,58 +1,65 @@
...
+    side1 == side2 && side2 == side3
+  end
+
+  def isosceles?
+    [side1, side2, side3].uniq.length == 2
+  end
 
+  def scalene?
+    ! (equilateral? || isosceles?)
+  end
```
Comment (zsyed1) [inline triangle_facts/triangle_facts.rb]: Extra space after the exclamation point in line 32

File: triangle_facts/triangle_facts.rb [hunk 2]
```diff
@@ -1,58 +1,65 @@
...
+
+  def recite_facts
+    display_triangle_type
+
+    angles = calculate_angles
+    puts 'The angles of this triangle are ' + angles.join(',')
+
+    puts 'This triangle is also a right triangle.' if angles.include? 90
+    puts ''
```
Comment (zsyed1) [inline triangle_facts/triangle_facts.rb]: Line 17: Put parentheses around method arguments: `if angles.include?(90)`.
Also, you can remove the blank line 16. 
Line 18: You can just do `puts` to write a newline. No need for empty string.

File: triangle_facts/triangle_facts.rb [hunk 3]
```diff
@@ -1,58 +1,65 @@
...
+    radians_to_degrees(Math.acos((b**2 + c**2 - a**2) / (2.0 * b * c)))
+  end
+
+  def calculate_angles
+    [
+      angle_in_degrees(side1, side2, side3),
+      angle_in_degrees(side2, side3, side1),
+      angle_in_degrees(side3, side1, side2),
+    ]
+  end
```
Comment (zsyed1) [inline triangle_facts/triangle_facts.rb]: Well done! Creating an `angle_in_degrees` method is a good move.

File: triangle_facts/triangle_facts.rb [hunk 4]
```diff
@@ -1,58 +1,65 @@
...
+  [5, 12, 13],
 ]
-triangles.each { |sides|
-	tri = Triangle.new(*sides)
-	tri.recite_facts
-}
+
+triangles.each do |sides|
+  Triangle.new(*sides).recite_facts
+end
```
Comment (zsyed1) [inline triangle_facts/triangle_facts.rb]: Solid work! This is done.

File: triangle_facts/triangle_facts.rb [hunk 5]
```diff
@@ -1,58 +1,65 @@
...
+    @side1 = side1
+    @side2 = side2
+    @side3 = side3
+  end
+
+  def recite_facts
+    display_triangle_type
+
+    angles = calculate_angles
+    puts 'The angles of this triangle are ' + angles.join(',')
```
Comment (djkotowski) [inline triangle_facts/triangle_facts.rb]: Use `<<` to concatenate and modify the leftmost string.

**Pull request closed without merging**